python/ray/llm/tests/serve/cpu/deployments/test_prefix_tree.py

```diff
@@ -0,0 +1,367 @@
+import pytest
+import time
+import ray
+from ray import serve
+
+from ray.llm._internal.serve.deployments.routers.prefix_tree import PrefixTree
+
+
+@pytest.fixture(scope="module", autouse=True)
+def serve_instance():
+    # Start Ray and Serve once per test module
+    ray.init(ignore_reinit_error=True)
+    serve.start(detached=True)
+    yield
+    serve.shutdown()
+    ray.shutdown()
+
+
+@pytest.mark.asyncio
+async def test_add_tenant():
+    """Test adding tenants to the tree."""
+    tree = serve.run(PrefixTree.bind())
```

Comment:
I would make PrefixTree a datastucture that for test purposes is not a deployment or ray actor. See how LLMServer is implemented for example.